core/src/db/engine/ExecutionEngineImpl.cpp

```diff
@@ -636,6 +641,38 @@ ExecutionEngineImpl::CopyToFpga() {
     return Status::OK();
 }
 
+Status
+ExecutionEngineImpl::CopyToApu(uint32_t row_count) {
+#ifdef MILVUS_APU_VERSION
+
+    auto cache_index_ =
+    std::static_pointer_cast<knowhere::VecIndex>(cache::FpgaCacheMgr::GetInstance()->GetItem(location_));
+    bool already_in_cache = (cache_index_ != nullptr);
+    //auto apu = Fpga::ApuInst::getInstance();
+    //bool isLoadNeeded =apu->isLoadNeeded(location_); //(index_ == nullptr);
+    if (!already_in_cache) {
+
+        cache::FpgaCacheMgr::GetInstance()->ClearCache(); // clear cache to support cache switch .
+
```

Comment:
It seems that you want to use FPGA cache to manage the APU data.
Why not to implement an APU cache?

Reply:
Hi Shengjun , 

yes we are using FpgaCach to check the current Apu loded data set . 
since APU is an FPGA device we wanted to reuse the code . 

please advise if this change is needed ,

Reply:
Different xxxCacheMgrs have different cache sizes, which are specified by the configuration file.
Previously, the model supported by FPGA was u250, which is with the same cache size as CPU.
Is the cache size of Apu different? If so, it is recommended to write another class named ApuCacheMgr.
Or the cache module can be reserved to implement later.